File: php/printDuplicateCharacters.php
<?php
  /*
  Print duplicate characters in a string.
  If string is "Javascript", output would be "a"

  Created on March 7, 2020
  @author: Charles Newman (https://github.com/26point2Newms)
  */

  $srcString = "hippopotomonstrosesquippedaliophobia";

  printf("\n\tDuplicate characters in '%s' are:\n", $srcString);
  print_r(findDuplicateChars($srcString));

  printf("\n\tDuplicate characters with their count in '%s' are:\n", $srcString);
  print_r(findDuplicateCharsWithCount($srcString));


  function findDuplicateCharsWithCount($srcStr)
  {
    $dupChars = [];
    $charsRead = [];
    $charsReadDictionary = [];

    $srcStrArray = str_split($srcStr);

    foreach ($srcStrArray as $char)
    {
      // Look in the list of characters we've read and add it to our duplicate list if it's not already in there
      if (count($charsRead) > 0 && in_array($char, $charsRead))
      {
        if (!in_array($char, $dupChars))
        {
           $dupChars[] = $char;

           // Set it's count to 2 because this is the 2nd time we've read that character
           $charsReadDictionary[$char] = 2;
        }
        else
          $charsReadDictionary[$char] += 1;
      }
      else
        $charsRead[] = $char;
    }
    
    return $charsReadDictionary;
  }
  
  function findDuplicateChars($srcStr)
  {
    $dupCharsList = [];
    $charsRead = [];
  
    $srcStrArray = str_split($srcStr);

    foreach ($srcStrArray as $char)
    {
      // Look in the characters list we've read and add it to our duplicate list if it's not already there
      if (in_array($char, $charsRead) && !in_array($char, $dupCharsList))
        $dupCharsList[] = $char;
      else
        $charsRead[] = $char;
    }
  
    return $dupCharsList;
  }

?>

File: python/printDuplicateCharacters.py
'''
Print duplicate characters in a string

If string is "Javascript", output would be "a"

Created on Jan 11, 2017

@author: Charles Newman (https://github.com/26point2Newms)
'''

def findDuplicateCharsWithCount(srcStr):
    dupCharsList = []
    charsRead = []
    charsReadDict = {}  # Dictionary for keeping the duplicate character and the number of times we've seen it
    
    for c in srcStr:
        # Look in the characters we've read list and add it to our dup list if it's not already in there
        if c in charsRead and len(charsRead) > 0:
            if not c in dupCharsList:
                dupCharsList.append(c)
                # set it's count to 2 because this is the 2nd time we've read that character
                charsReadDict[c] = 2
            else:
                charsReadDict[c] = charsReadDict[c] + 1
        else:
            charsRead.append(c)
            
    return charsReadDict


def findDuplicateChars(srcStr):
    dupCharsList = []
    charsRead = []
    
    for c in srcStr:
        # Look in the characters we've read list and add it to our dup list if it's not already in there
        if c in charsRead and not c in dupCharsList:
            dupCharsList.append(c)
        else:
            charsRead.append(c)
            
    return dupCharsList
    
    
def main():
    srcString = "hippopotomonstrosesquippedaliophobia"
    print(str(findDuplicateChars(srcString.lower())))
    print(str(findDuplicateCharsWithCount(srcString.lower())))
    
if __name__ == '__main__':
    main()
    